lib/livebook/runtime/erl_dist/runtime_server.ex

```diff
@@ -1023,4 +1031,94 @@ defmodule Livebook.Runtime.ErlDist.RuntimeServer do
   end
 
   defp intellisense_node(_node), do: node()
+
+  defp build_identifiers(metadata) do
+    identifiers_defined = metadata.identifiers_defined
+
+    Enum.reduce_while(identifiers_defined, {:ok, %{}}, fn
+      {{:module, module}, _}, {:ok, acc} ->
+        with {^module, bin, _} <- :code.get_object_code(module),
+             {:ok, debug_info} <- beam_lib_chunks(bin, :debug_info),
+             {:ok, abstract_code} <- beam_lib_chunks(bin, :abstract_code),
+             {:ok, types} <- Code.Typespec.fetch_types(module) do
+          {:cont,
+           {:ok,
+            Map.put(
+              acc,
+              module_name(module),
+              build_module_identifier(abstract_code)
+              |> build_function_identifiers(module, debug_info)
+              |> build_type_identifiers(module, types)
+            )}}
+        else
+          :error -> {:halt, :error}
+        end
+
+      _, acc ->
+        {:cont, acc}
+    end)
+  end
+
+  defp beam_lib_chunks(bin, key) do
+    case :beam_lib.chunks(bin, [key]) do
+      {:ok, {_, kw}} -> {:ok, Keyword.fetch!(kw, key)}
+      {:error, :beam_lib, _} -> :error
+    end
+  end
+
+  defp build_module_identifier({:raw_abstract_v1, annotations}) do
+    {:attribute, anno, :module, module} =
+      Enum.find(annotations, &match?({:attribute, _, :module, _}, &1))
+
+    %{
+      kind: :module,
+      module: module,
+      name: module_name(module),
+      arity: nil,
+      line: :erl_anno.line(anno)
+    }
+  end
+
+  defp build_module_identifier(_), do: %{}
```

Comment:
Would returning an empty map break the front end? Maybe it is best to skip this module altogether if we can't find its line?